sparse/client.go

```diff
@@ -79,41 +72,17 @@ func syncFile(localPath string, addr TCPEndPoint, remotePath string, timeout int
 	encoder := gob.NewEncoder(conn)
 	decoder := gob.NewDecoder(conn)
 
-	// Use unix time as hash salt
-	salt := make([]byte, binary.MaxVarintLen64)
-	binary.PutVarint(salt, time.Now().UnixNano())
-	status := sendSyncRequest(encoder, decoder, remotePath, size, salt)
-	if !status {
-		err = fmt.Errorf("Sync request to %v failed", remotePath)
-		log.Error(err)
+	ack, err := sendSyncRequest(encoder, decoder, remotePath, fileSize)
```

Comment:
Make this two fields as a struct, and any function call related can be written as this object's method. Trying to get something more object orientated. E.g. you can create client object.